pkg/metadata/inventories/host_metadata.go

```diff
@@ -90,21 +90,23 @@ func fetchFromMetadata(key string, metadata AgentMetadata) string {
 func getHostMetadata() *HostMetadata {
 	metadata := &HostMetadata{}
 
-	cpuInfo, warnings, err := cpuGet()
+	cpuInfo := cpuGet()
+	_, warnings, err := cpuInfo.AsJSON()
 	if err != nil {
 		logErrorf("Failed to retrieve cpu metadata from gohai: %s", err) //nolint:errcheck
 	} else {
 		logWarnings(warnings)
 
-		metadata.CPUCores = cpuInfo.CpuCores
-		metadata.CPULogicalProcessors = cpuInfo.CpuLogicalProcessors
-		metadata.CPUVendor = cpuInfo.VendorId
-		metadata.CPUModel = cpuInfo.ModelName
-		metadata.CPUModelID = cpuInfo.Model
-		metadata.CPUFamily = cpuInfo.Family
-		metadata.CPUStepping = cpuInfo.Stepping
-		metadata.CPUFrequency = cpuInfo.Mhz
-		metadata.CPUCacheSize = cpuInfo.CacheSizeBytes
+		metadata.CPUCores, _ = cpuInfo.CPUCores.Value()
+		metadata.CPULogicalProcessors, _ = cpuInfo.CPULogicalProcessors.Value()
+		metadata.CPUVendor, _ = cpuInfo.VendorID.Value()
```

Comment:
What do you think about having a `ValueOrDefault` method as relying on the value when there is an error is uncommon.

Reply:
Good idea, I'll implement that in a follow-up PR as it also changes code in other gohai packages